mm2src/mm2_main/src/lp_swap/swap_watcher.rs

```diff
@@ -194,10 +180,23 @@ impl State for ValidateTakerPayment {
             )));
         }
 
+        let taker_payment_hex_fut = watcher_ctx
```

Comment:
Please consider doing this before `wait_for_confirmations` - you can then use the retrieved hex in older `wait_for_confirmations` and remove `wait_for_confirmations_by_hash`.

Reply:
I can do it, but then I also have to do it inside a loop because at this point the taker payment might not be on the chain yet.

Reply:
Even if transaction is not confirmed, it should be still returned by coin RPC. Though, it might take some time for it to be propagated to all nodes' mempool. Have you encountered this in integration tests?

Reply:
No in integration tests everything was working fine. I still thought it would be a good idea to put this in a loop in case there's delays.